src/storage/temp_table.cpp

```diff
@@ -0,0 +1,133 @@
+
+//===----------------------------------------------------------------------===//
+//
+//                         PelotonDB
+//
+// temp_table.cpp
+//
+// Identification: /peloton/src/storage/temp_table.cpp
+//
+// Copyright (c) 2015, 2016 Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "storage/temp_table.h"
+
+#include "catalog/schema.h"
+#include "common/exception.h"
+#include "common/logger.h"
+#include "storage/tile_group.h"
+#include "storage/tile_group_header.h"
+#include "storage/tuple.h"
+
+namespace peloton {
+namespace storage {
+
+TempTable::TempTable(const oid_t &table_oid, catalog::Schema *schema,
+                     const bool own_schema)
+    : AbstractTable(table_oid, schema, own_schema) {
+  // We only want to instantiate a single TileGroup
+  AddDefaultTileGroup();
+}
+
+TempTable::~TempTable() {
+  // Nothing to see, nothing to do
+}
+
+ItemPointer TempTable::InsertTuple(const Tuple *tuple,
+                                   concurrency::Transaction *transaction,
```

Comment:
The `transaction` and the `index_entry_ptr` should be tagged as `UNUSED_ATTRIBUTE`, otherwise it will failed the release build.